include/bluetooth/services/dfu_smp_c.rst

```diff
@@ -0,0 +1,77 @@
+.. _dfu_smp_c_readme:
+
+GATT DFU SMP Service Client
+###########################
+
+This module implements a Simple Management Protocol (SMP) Service Client that can be used in the context of Device Firmware Updates (DFU).
+SMP is a basic transfer encoding for use with the `MCUmgr`_ management protocol.
+See `SMP over Bluetooth`_ for the service specification.
+
+The SMP Client module can be used to interact with Zephyr's :ref:`zephyr:smp_svr_sample`.
+
+The SMP Client implements only the service.
+It does not provide any functionality to process or interpret SMP commands and responses.
+
+Service UUID
+************
+
+The 128-bit service UUID is 8D53DC1D-1DB7-4CD3-868B-8A527460AA84 (16-bit offset: 0x0001).
+
+Characteristics
+***************
+
+This service has one characteristic that is used for both requests and responses.
+
+SMP Characteristic (DA2E7828-FBCE-4E01-AE9E-261174997C48)
+=========================================================
+
+Write Without Response
+   * Write an SMP command with data.
+
+Notify
+   * Response to the SMP command is sent using notification.
+   * Multiple notifications may be generated if the response does not fit in a single MTU.
+
+Usage
+*****
+
+.. note::
+   Do not access any of the values in the :cpp:type:`bt_gatt_dfu_smp_c` object structure directly.
+   All values that should be accessed have accessor functions.
+   The reason that the structure is fully defined is to allow the application to allocate the memory for it.
+
+MTU negotiation
+===============
+
+The current DFU SMP Server implementation in the :ref:`zephyr:smp_svr_sample` requires the whole message to be sent in one transfer.
+For most operations, this requires a bigger MTU size than the default one.
+Therefore, you must increase the default MTU size for the connection.
```

Comment:
Is that correct? How is that done? (It's sufficient if we can just link somewhere.)

Reply:
Default MTU size is the size default in Bluetooth specification. MTU - Message Transfer Unit.
The protocol starts with the default size of 23 bytes. To change this default MTU size the MTU exchange process has to be initiated. On Zephyr we init it by bt_gatt_exchange_mtu.
Then our client sends MTU size it can support and the server sends it too. The smaller value is choosen for the future communication.

Reply:
Changed to "For most operations, this requires a bigger MTU size than the default one, which is negotiated in the MTU exchange process (see bt_gatt_exchange_mtu()). " - ok?

Reply:
Well... In my opinion in this sentence we can read it in two ways:
- Bigger MTU size is negotiated in the...
- Default MTU size is negotiated in the... but we need bigger in some magic way.

Maybe:
"Most operations require a bigger MTU size than the default one. This requires MTU negotiation in the MTU exchange process (see bt_gatt_exchange_mtu()). "